File: BITEC/current lesson/oop/src/KlassenUndBeziehungen/HundeBesitzer.java
package KlassenUndBeziehungen;

public class HundeBesitzer extends Mensch {

    // Felder
    private boolean hatHundeFuehrerschein;

    // Beziehungen
    private Hund[] hunde;

    // Konstruktor
    public HundeBesitzer(String name, double happiness, int alter, boolean hatHundeFuehrerschein, int capacity) {
        super(name, happiness, alter);
        this.hatHundeFuehrerschein = hatHundeFuehrerschein;
        this.hunde = new Hund[capacity];
    }

    // Methoden
    public void buersten() {

    }

    public void gassiGehen() {

    }

    public void fuettern() {

    }

    // get und set methoden
    public boolean isHatHundeFuehrerschein() {
        return hatHundeFuehrerschein;
    }

    public void setHatHundeFuehrerschein(boolean hatHundeFuehrerschein) {
        this.hatHundeFuehrerschein = hatHundeFuehrerschein;
    }

    public Hund[] getHunde() {
        return hunde;
    }

    public void setHunde(Hund[] hunde) {
        this.hunde = hunde;
    }
}

File: C#/C#.NET/modul_1_grundlagen/LiveCoding/L04KlassenUndBeziehungen/HundeBesitzer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LiveCodingKlassen;

public class HundeBesitzer : Mensch
{
    // Felder
    private int capacity;
    private bool hatHundeFuehrerschein;

    // Hat - Beziehungen
    private List<Hund> hunde = new List<Hund>();

    // Methoden:
    // - Konstruktor
    public HundeBesitzer(string name, int alter, double happiness, int capacity, Hund meinErsterHund) : base(name, alter, happiness)
    {
        this.capacity = capacity;
        meinErsterHund.SetBesitzer(this);
    }

    // - Verhaltensmethoden

    // - Get-Set Methoden
    public void AddHund(Hund hund)
    {
        hunde.Add(hund);
    }

    public bool BesitztHund(Hund hund)
    {
        return hunde.Contains(hund);
    }
}